arangod/Aql/DocumentProducingHelper.h

```diff
@@ -0,0 +1,52 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2019 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Heiko Kernbach
+////////////////////////////////////////////////////////////////////////////////
+
+#ifndef ARANGOD_AQL_DOCUMENT_PRODUCING_HELPER_H
+#define ARANGOD_AQL_DOCUMENT_PRODUCING_HELPER_H 1
+
+#include "Aql/AqlItemBlock.h"
+#include "Aql/ExecutionNode.h"
+#include "Aql/InputAqlItemRow.h"
+#include "Aql/OutputAqlItemRow.h"
+#include "Basics/Common.h"
+
+#include <velocypack/Slice.h>
+
+namespace arangodb {
+namespace aql {
+
+class DocumentProducingHelper {
```

Comment:
Why is a single static function in a class?

Reply:
To be extendable in the future? What would be a better approach?

Reply:
I am not sure in which way it needs to be extendable but, to me it really looks like the static function should rather be a free-standing-function.